### ISSUE TITLE
Use PRB0t in the browser

### ISSUE THREAD

**Issue by j-rewerts** [PRB0t/PRB0t#39]
One of the use cases detailed in the readme is for adding live editing to static sites. I'm guessing this means PRB0t should be able to be used in the browser. It seems like CORS prohibits it right now.

Check out [this](https://codepen.io/j-rewerts/pen/eoBvmG) CodePen for what I'm imagining.

**Comment by whtiehack** [comment]
@j-rewerts 
You can publish the app by yourself.
And you can refer to this,set the headers as this.
```
    res.setHeader("Access-Control-Allow-Origin", "*");
    res.setHeader("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
    res.setHeader("Access-Control-Allow-Methods", "POST,GET");
    res.setHeader('Content-Type', 'application/json;charset=utf-8');
```

**Comment by j-rewerts** [comment]
Yeah that is what I ended up doing. @whtiehack Are you interested in having this functionality in PRB0t?

**Comment by whtiehack** [comment]
There is no such requirement yet.
Now just  `new PR()` and  `pr.send()`.   :)

**Comment by whtiehack** [comment]
My requirement is when create branch, need to create branch from the upstream repo.
Now the PRB0t can only create branch from fork repo branch .  
Now I'm researching the Github API and see how it works.

**Comment by M3kH** [comment]
The reason why is a fork is because the user is `PRB0T` but I can add this functionality by adding two options.
- [X] ~Add OAuth Token (So you can use your own custom user from the request it self). I will prefer to avoid this and just keep the option to self hosting option.~ *This is already there*
- [ ] Add the option to the constructor to choose branch rather than fork.

@whtiehack Would that work for you?

**Comment by whtiehack** [comment]
In fact, you can set your own token now.
Because you has a token parameter in the PullRequest contruct.

https://github.com/PRB0t/PRB0t/blob/ccaacfa98098c4e84c5d847f8bd2c17383b84f28/src/PullRequest.js#L6-L16

Just need to set up like this:
```
{
  "user": "<github-user>",
  "repo": "<github-repo>",
  "title": "<pull-request-title>",
  "description": "<pull-request-description>",
  "commit": "<commit-description>",
  "files": [
  	{"path": "<file-path>", "content": "<file-content>"}
  ],
  "token":"YOU TOKEN here"
}
```

> Add the option to the constructor to choose branch rather than fork.
+1.

For example, the original repository: `PRB0t/PRB0t`, my FORK repository `whtiehack/PRB0t`, when I submit the PR,I can choose create a PR branch with `PRB0t/PRB0t` instead of ` whtiehack/PRB0t`.

@M3kH  ,thanks for you tools again.

**Comment by j-rewerts** [comment]
@M3kH Let me know if you want my upstream CORS code. I can submit as a PR. If not, feel free to close this issue.

**Comment by M3kH** [comment]
@j-rewerts if doesn't bother you too much I'm just curious which changes you apply for make it CORS compatible.
Although I rather not implement that in my published endpoint to avoid abuse of it.

**Comment by j-rewerts** [comment]
I did it in [this](https://github.com/j-rewerts/PRB0t/pull/2) PR.

**Comment by j-rewerts** [comment]
It would be nice to somehow make this configurable at deploy time for PRBot. That way, I don't have to maintain a hard fork of PR Bot. @M3kH Would be open to that? You could keep your no-CORS deployment and I could just pass in config options for my CORS deployment?

**Comment by M3kH** [comment]
@j-rewerts Is there a way for you to test this branch https://github.com/PRB0t/PRB0t/pull/41 ?
I don't use `now` for deployment, so would be glad if you can :-)

**Comment by j-rewerts** [comment]
I've actually never heard of now. I deployed using Heroku.

**Comment by j-rewerts** [comment]
Closed with #41.
